src/NLog.Web/NLogRequestLoggingModule.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.Web;
+
+namespace NLog.Web
+{
+    /// <summary>
+    /// HttpModule that writes all requests to Logger named "RequestLogging"
+    /// </summary>
+    public class NLogRequestLoggingModule : IHttpModule
+    {
+        private static readonly NLog.Logger Logger = NLog.LogManager.GetLogger("RequestLogging");
+
+        void IHttpModule.Init(HttpApplication context)
+        {
+            context.EndRequest += LogHttpRequest;
+        }
+
+        private void LogHttpRequest(object sender, EventArgs e)
+        {
+            Exception exception = null;
+            int statusCode = 400;
+
+            try
+            {
+                exception = HttpContext.Current?.Server?.GetLastError();
+                statusCode = HttpContext.Current?.Response?.StatusCode ?? 400;
```

Comment:
I don't get this default of 400. Maybe we should make it at least configurable. or write below `statusCode >= 400 or statuscode is null` (pseudecode)

Reply:
Not a ASP.NET developer, so clue when a HttpStatusCode should signal a warning. Just assumed anything above 400 was a serious error (and not just a redirection). But I can also see other definitions: https://github.com/microsoft/referencesource/blob/5697c29004a34d80acdaf5742d7e699022c64ecd/System/net/System/Net/Http/HttpResponseMessage.cs#L117

Reply:
Removed 400 as default and instead signal warning, when StatusCode < 100